src/models/queues.ts

```diff
@@ -289,6 +293,18 @@ export interface QueueDocument extends Queue, mongoose.Document<mongoose.Types.O
      * Returns `true` if the Queue is Empty
      */
     isEmpty(): boolean,
+    /**
+     * Calls the callback-function passed as a parameter when a user joins the previously empty queue.
+     * The callback function will be associated with the passed discord id.
+     * @param discord_id The discord id / user associated with the callback function.
+     * @param callback_function The function that is called in the notification case.
+     */
+    subscribeOnJoinWhenEmpty(discord_id: string, callback_function: () => void): boolean,
+    /**
+     * Cancels the notification of the user with the specified discord id when a user joins the previously empty queue.
+     * @param discord_id The discord id / user associated with the callback function.
+     */
+    unsubscribeOnJoinWhenEmpty(discord_id: string): boolean,
```

Comment:
Alles was aus der DB liest oder in die db Schreibt muss asyncron sein, hier also jeweils Rückgabetyp `Promise<boolean>`